analyzers/tests/SonarAnalyzer.UnitTest/TestCases/SymbolicExecution/Roslyn/ConditionEvaluatesToConstant.cs

```diff
@@ -3429,3 +3429,14 @@ void Test()
         }
     }
 }
+
+// Reproducer for https://github.com/SonarSource/sonar-dotnet/issues/8445
+public class Repro_8445
+{
+    private static readonly int readonlyField = 42;
+    private void Test()
+    {
+        if (readonlyField == 42)    // FN
+            Console.WriteLine();
+    }
```

Comment:
Add tests for additional types:

```cs
public class Repro_8445
{
    private static readonly bool readonlyFieldBool = true;
    private static readonly int readonlyFieldInt = 42;
    private static readonly string readonlyFieldString = "ABC";

    public void Test()
    {
        if (readonlyFieldBool)                   // FN
            Console.WriteLine();

        if (!readonlyFieldBool)                  // FN
            Console.WriteLine();

        if (readonlyFieldInt == 42)              // FN
            Console.WriteLine();

        if (readonlyFieldInt != 42)              // FN
            Console.WriteLine();

        if (readonlyFieldString == "ABC")        // FN
            Console.WriteLine();

        if (readonlyFieldString != "ABC")        // FN
            Console.WriteLine();

        if (AnotherClass.readonlyFieldInt == 42) // FN
            Console.WriteLine();

        if (AnotherClass.readonlyFieldInt != 42) // FN
            Console.WriteLine();
    }

    public class AnotherClass
    {
        public static readonly int readonlyFieldInt = 42;
    }
}
```

Reply:
This is not so much about the type but more about the field, so I feel some of the tests would be redundant. But I will add a test for a reference type and one for a field in another class. Thanks for the suggestion!